crypto/pkcs12/p12_mutl.c

```diff
@@ -238,9 +363,119 @@ int PKCS12_set_mac(PKCS12 *p12, const char *pass, int passlen,
     return 1;
 }
 
-/* Set up a mac structure */
-int PKCS12_setup_mac(PKCS12 *p12, int iter, unsigned char *salt, int saltlen,
-                     const EVP_MD *md_type)
+static int pkcs12_pbmac1_key_gen(const char *pass, int passlen,
+                                 unsigned char *salt, int saltlen,
+                                 int id, int iter, int keylen,
+                                 unsigned char *out,
+                                 const EVP_MD *md_type)
+{
+    return PKCS5_PBKDF2_HMAC(pass, passlen, salt, saltlen, iter,
+                           md_type, keylen, out);
+}
+
+static int pkcs12_mdnid2hmacnid(int mdnid)
+{
+    int hmac_nid = NID_undef;
+
+    switch(mdnid) {
+        case NID_sha1:                    hmac_nid = NID_hmacWithSHA1; break;
```

Comment:
again, this looks like code that should exist somewhere else already...

Reply:
Unfortunately no.